File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/application/StructuredExtractor.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.domain.model.ExecutionPlan;
import com.gonezo.multiplatform.audioextraction.domain.model.FieldCandidate;
import com.gonezo.multiplatform.audioextraction.domain.model.Transcript;
import com.gonezo.multiplatform.audioextraction.domain.schema.OutputSchema;
import java.util.List;
import java.util.Map;

public interface StructuredExtractor {
  Map<String, List<FieldCandidate>> extract(Transcript transcript, ExecutionPlan plan, OutputSchema schema);
}

File: core/bin/main/com/gonezo/audioextraction/application/pipeline/StructuredExtractor.kt
package com.gonezo.audioextraction.application.pipeline

import com.gonezo.audioextraction.domain.model.ExecutionPlan
import com.gonezo.audioextraction.domain.model.FieldCandidate
import com.gonezo.audioextraction.domain.model.Transcript
import com.gonezo.audioextraction.domain.schema.OutputSchema

interface StructuredExtractor {
  fun extract(transcript: Transcript, plan: ExecutionPlan, schema: OutputSchema): Map<String, List<FieldCandidate>>
}
